eclair-core/src/test/scala/fr/acinq/eclair/blockchain/watchdogs/BlockchainWatchdogSpec.scala

```diff
@@ -65,12 +70,14 @@ class BlockchainWatchdogSpec extends ScalaTestWithActorTestKit(ConfigFactory.loa
     val eventListener = TestProbe[DangerousBlocksSkew]()
     system.eventStream ! EventStream.Subscribe(eventListener.ref)
     val blockTimeout = 5 seconds
-    val watchdog = testKit.spawn(BlockchainWatchdog(Block.TestnetGenesisBlock.hash, 1 second, blockTimeout))
+    val nodeParams = TestConstants.Alice.nodeParams.copy(chainHash = Block.TestnetGenesisBlock.hash)
+    val watchdog = testKit.spawn(BlockchainWatchdog(nodeParams, 1 second, blockTimeout))
 
     watchdog ! WrappedCurrentBlockCount(500000)
     assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
     assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
     assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
+    assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
```

Comment:
Why is there a third event? There are only 3 watchdogs.

Reply:
That's because you disabled blockcypher. There are in fact only two watchdogs enabled in that test (`blockstream.info` and `mempool.space`). The checks should instead be (the commented check is for the disabled `blockcypher` watchdog):

```scala
    watchdog ! WrappedCurrentBlockCount(500000)
    assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
    assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
    // assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
    eventListener.expectNoMessage(100 millis)

    // If we don't receive blocks, we check blockchain sources.
    assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
    assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
    // assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
    eventListener.expectNoMessage(100 millis)

    // And we keep checking blockchain sources until we receive a block.
    assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
    assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
    // assert(eventListener.expectMessageType[DangerousBlocksSkew].recentHeaders.currentBlockCount === 500000)
    eventListener.expectNoMessage(100 millis)
```

The issue was that if you expected a third message, you were in fact waiting for the `blockTimeout`, and that triggered two messages (since there are two watchdogs).